internal/server/server.go

```diff
@@ -32,32 +31,62 @@ func Run(opts ...Option) error {
 		cfg.Logger = hclog.L()
 	}
 
-	// Setup our run group since we're going to be starting multiple
-	// goroutines for all the servers that we want to live/die as a group.
-	var group run.Group
-
-	// We first add an actor that just returns when the context ends. This
-	// will trigger the rest of the group to end since a group will not exit
-	// until any of its actors exit.
-	ctx, cancelCtx := context.WithCancel(cfg.Context)
-	cfg.Context = ctx
-	group.Add(func() error {
-		<-ctx.Done()
-		return ctx.Err()
-	}, func(error) { cancelCtx() })
+	log := cfg.Logger.Named("server")
```

Comment:
Another thing to consider is making another sub-logger `log.named("grpc")` (and HTTP) since I noticed your log messages in each start with "HTTP Server" or "GRPC Server" a lot which is a smell to me you want a named logger.

Reply:
I had the same suggestion. I added a bunch of suggestions for logging messages to be more explicit about which server is doing what. But if the grpc and http server is formatted like ☝🏻 then I think most of those suggestions won't be needed.

Reply:
I removed those review comments, we should just structure each servers logger to be named.

Reply:
Hmm, I am creating and using named [http](https://github.com/hashicorp/waypoint/pull/1761/files#diff-5f2a899e8c4b83b44a53b5bea4e1a3129c4c9cadeab0331c3d3e6e6affac42a2R26) and [grpc](https://github.com/hashicorp/waypoint/pull/1761/files#diff-85d6db9d4d33f7064a02bbe45907ffdaca1cd1382bbfdb5ab8c4a7e77841cea5R23) loggers in their respective files. Are you suggesting that I create those loggers earlier, here in server.go, and use them [here](https://github.com/hashicorp/waypoint/pull/1761/files#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23R59) and [here](https://github.com/hashicorp/waypoint/pull/1761/files#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23R79)?

Reply:
I would keep this logger, and instead of creating a new one in each function, pass this `log` through to the functions that need it and you can modify it using the `With` to give it the detail for which kind of server it is:

```golang
func newHttpServer(grpcServer *grpc.Server, log hclog.Logger, opts *options) *httpServer {
  log = log.With("http")
  // stuff
}
```

Then it should show up namespaced like `waypoint.server.http` or `waypoint.server.grpc` for example.

Reply:
Hmm, from my reading it looks like log.With is for adding key/value pairs. The logs do come through named the way I would expect in the current implementation - here's a snippet from shutdown:
```
2021-07-07T09:56:09.441-0500 [DEBUG] waypoint.server.http: stopping
2021-07-07T09:56:09.441-0500 [DEBUG] waypoint.server: HTTP server has exited: err="http: Server closed"
2021-07-07T09:56:09.442-0500 [DEBUG] waypoint.server.http: exited gracefully
2021-07-07T09:56:09.442-0500 [DEBUG] waypoint.server.grpc: stopping
2021-07-07T09:56:11.443-0500 [DEBUG] waypoint.server.grpc: stopping forcefully after waiting unsuccessfully for graceful stop
2021-07-07T09:56:11.443-0500 [DEBUG] waypoint.server: gRPC server has exited: err=<nil>
```

The parent logger is passed into the sub-functions as part of opts instead of as a standalone argument, which was the existing behavior.

Reply:
Ahh, alright! I wasn't sure if that was the case or if the namespace was lost when you create a new one off that options config. That makes sense to me then 👍🏻